src/realm/sync/instruction_replication.cpp

```diff
@@ -81,6 +82,14 @@ Instruction::Payload SyncReplication::as_payload(Mixed value)
             break;
         }
     }
+    if (type == type_Dictionary) {
+        // throw IllegalOperation("Cannot sync nested dictionary");
+        return Instruction::Payload(Instruction::Payload::Dictionary());
+    }
+    else if (type == type_List) {
+        // throw IllegalOperation("Cannot sync nested list");
+        return Instruction::Payload(Instruction::Payload::List());
+    }
```

Comment:
Did you intend to keep the commented out throw statements?

Reply:
We need to disallow syncing of nested collections until this is supported by the server. So at some point we should throw here,

Reply:
Oh, I see :+1:

Reply:
Shouldn't we throw now and then update it to return the payload when we have server support?